docs/source/api/apollo-server.mdx

```diff
@@ -457,203 +407,93 @@ The default value is `true`. Set this to `false` to use mocked resolvers only fo
+## `addPlugin`
+You can use the `addPlugin` method to add plugins before your server is started. This is helpful when you want to add a plugin that accepts your initialized `ApolloServer` instance itself, like so:
 
-Always call `await server.start()` *before* calling `server.applyMiddleware` and starting your HTTP server. This allows you to react to Apollo Server startup failures by crashing your process instead of starting to serve traffic.
+<MultiCodeBlock>
 
-If the only thing you are doing with your server is calling [`executeOperation`](../testing/testing/) on it (ie, you're not actually starting an HTTP server), you don't have to call `start()`; `executeOperation` will do that for you.
+```ts
+const server = new ApolloServer({
+  typeDefs,
+  plugins: [
+    makeFirstPlugin(),
+  ],
+});
 
-This method was optional in Apollo Server 2 but is required in Apollo Server 3.
+server.addPlugin(makeSecondPlugin(server));
+```
 
-### Triggered actions
+</MultiCodeBlock>
 
-The `start` method triggers the following actions:
-
-1. If your server is a [federated gateway](https://www.apollographql.com/docs/federation/managed-federation/overview/), it attempts to fetch its schema. If the fetch fails, `start` throws an error.
-2. Your server calls all of the [`serverWillStart` handlers](../integrations/plugins-event-reference/#serverwillstart) of your installed plugins. If any of these handlers throw an error, `start` throws an error.
+## `assertStarted`
 
+Framework integrations should call `server.assertStarted()` to ensure a server has started before accepting requests. See [Building integrations](../integrations/building-integrations) for more details.
 
 ## `stop`
 
 `ApolloServer.stop()` is an async method that tells all of Apollo Server's background tasks to complete. Specifically, it:
 
-- Calls and awaits all [`drainServer` plugin handlers](../integrations/plugins-event-reference/#drainserver). These should generally:
-  * Stop listening for new connections
-  * Closes idle connections (i.e., connections with no current HTTP request)
-  * Closes active connections whenever they become idle
-  * Waits for all connections to be closed
-  * After a grace period, if any connections remain active, forcefully close them.
-  If you're using the batteries-included `apollo-server` package, it does this by default. (You can configure the grace period with the [`stopGracePeriodMillis` constructor option](#stopgraceperiodmillis).) Otherwise, you can use the [drain HTTP server plugin](./plugin/drain-http-server/) to drain your HTTP server.
+- Calls and awaits all [`drainServer` plugin handlers](/apollo-server/integrations/plugins-event-reference/#drainserver). These should generally:
+  - Stop listening for new connections
+  - Closes idle connections (i.e., connections with no current HTTP request)
+  - Closes active connections whenever they become idle
+  - Waits for all connections to be closed
```

Comment:
I think the verbs here should match the plural "these" from "these should generally:"
```suggestion
  - Close idle connections (i.e., connections with no current HTTP request)
  - Close active connections whenever they become idle
  - Wait for all connections to be closed
```